pymeasure/instruments/siglenttechnologies/siglent_sds1072cml.py

```diff
@@ -0,0 +1,400 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2024 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import struct
+
+from pymeasure.instruments import Channel, Instrument
+from pymeasure.instruments.generic_types import SCPIMixin
+from pymeasure.instruments.validators import truncated_discrete_set, truncated_range
+
+
+class VoltageChannel(Channel):
+    """
+    ===========================================================
+    Implementation of a SIGLENT SDS1072CML Oscilloscope channel
+    ===========================================================
+    """
+
+    vertical_division = Channel.control(
+        "C{ch}:VDIV?",
+        "C{ch}:VDIV %s",
+        "Control the vertical sensitivity of a channel in V/divisions.",
+        validator=truncated_range,
+        values=[2e-3, 10],
+        get_process=lambda v: float(v.split(" ", 1)[-1][:-1]),
+        set_process=lambda v: "%.2eV" % v,
+    )
+
+    coupling = Channel.control(
+        "C{ch}:CPL?",
+        "C{ch}:CPL %s1M",
+        "Control the channel coupling mode.(DC or AC)",
+        validator=truncated_discrete_set,
+        values={"DC": "D", "AC": "A"},
+        map_values=True,
+        get_process=lambda v: v.split(" ", 1)[-1][0],
+    )
+
+    def get_waveform(self):
+        """Get the waveforms displayed in the channel as a tuple with elements:
+        - time: (1d array) the time in seconds since the trigger epoch for every voltage value in
+        the waveforms
+        - voltages: (1d array) the waveform in V
+
+        """
+        command = "C{ch}:WF? ALL"
+        self.write(command)
```

Comment:
Just a suggestion:
you can combine these two lines into one, as it won't get overly complex.
```suggestion
        self.write( "C{ch}:WF? ALL")
```